model/project_ref.go

```diff
@@ -2542,6 +2548,39 @@ func (p *ProjectRef) GetActivationTimeForVariant(variant *BuildVariant, versionC
 	return versionCreateTime, nil
 }
 
+// CheckAndUpdateAutoRestartLimit checks if auto restarting a task for a project is allowed given
+// the global per-project daily auto restarting limit, and updates relevant timestamp and counter used
+// to track the project's usage.
+func (p *ProjectRef) CheckAndUpdateAutoRestartLimit(maxDailyAutoRestarts int) error {
+	if maxDailyAutoRestarts == 0 {
+		return nil
+	}
+	var update bson.M
+	now := time.Now()
+	oneDayAgo := now.Add(-1 * 24 * time.Hour)
```

Comment:
Since this is pretty generic and potentially reusable, can we move it into a util function?